src/components/ResourceList/ResourceList.tsx

```diff
@@ -480,6 +480,7 @@ class ResourceListInner extends React.Component<CombinedProps, State> {
 
     const headerMarkup = !showEmptyState &&
       (showHeader || needsHeader) &&
+      showHeader !== false &&
```

Comment:
Ah I missed this check the first time around.

```suggestion
      (showHeader !== false || needsHeader) &&
```

Can we merge this with the line above?

Reply:
By merging it like that `(showHeader || needsHeader)` would come back as true when `showHeader` is `undefined` which I don't think is what we want. Currently it would come back as `false` when `showHeader` is falsy and we then check if isn't set to `false` explicitly (`showHeader !== false`) as in that case we don't want the header to render.

Reply:
Ugh, right. Disregard

Reply:
I do think though that this condition is really confusing. Outside the scope of this PR but it might benefit from some cleanup.